agent/xds/proxysupport/proxysupport.go

```diff
@@ -12,3 +14,28 @@ var EnvoyVersions = []string{
 	"1.22.5",
 	"1.21.5",
 }
+
+// UnsupportedEnvoyVersions lists any unsupported Envoy versions (mainly minor versions) that fall
+// within the range of EnvoyVersions above.
+// For example, if patch 1.21.3 (patch 3) had a breaking change, and was not supported
+// even though 1.21 is a supported major release, you would then add 1.21.3 to this list.
+// This list will be empty in most cases.
+//
+// see: https://www.consul.io/docs/connect/proxies/envoy#supported-versions
+var UnsupportedEnvoyVersions = []string{}
+
+// GetMaxEnvoyMajorVersion grabs the first value in EnvoyVersions and strips the patch number off in order
+// to return the maximum supported Envoy Major version
+// For example, if the input string is "1.14.1", the function would return "1.14".
+func GetMaxEnvoyMajorVersion() string {
+	s := strings.Split(EnvoyVersions[0], ".")
```

Comment:
Is it reasonable for us to assume that the 0th element of the list is the max major version?

I don't see a comment above `EnvoyVersions` stating that the array is expected to be sorted from newest to oldest major version. I think we should either document that expectation, or have the get min/max code actually iterate through the array to pull out the actual min/max (not assuming that `EnvoyVersions` is in order).

Reply:
Yeah, the array has this comment:
```
// EnvoyVersions lists the latest officially supported versions of envoy.
//
// This list must be sorted by semver descending. Only one point release for
// each major release should be present.
//
// see: https://www.consul.io/docs/connect/proxies/envoy#supported-versions
```

Some other parts of the code were assuming order, so I did the same

Reply:
I added a test to make sure it is in descending order as per the instructions